src/frontend/src/lib/services/address.services.ts

```diff
@@ -69,22 +72,43 @@ const loadTokenAddress = async <T extends Address>({
 	return { success: true };
 };
 
+// We use the Testnet address for Regtest.
+type TokenIdBtcPublicNetwork = typeof BTC_MAINNET_TOKEN_ID | typeof BTC_TESTNET_TOKEN_ID;
+type BtcPublicNetwork = Exclude<BitcoinNetwork, 'regtest'>;
+const bitcoinStoreMapper: Record<BtcPublicNetwork, AddressStore<BtcAddress>> = {
+	mainnet: btcAddressMainnetStore,
+	testnet: btcAddressTestnetStore
+};
+
+const bitcoinNetworkMapper: Record<BtcPublicNetwork, SignerBitcoinNetwork> = {
+	mainnet: { mainnet: null },
+	testnet: { testnet: null }
+};
+
+const idbBtcAddressMapper: Record<
+	BtcPublicNetwork,
+	(params: SetIdbAddressParams<BtcAddress>) => Promise<void>
+> = {
+	mainnet: setIdbBtcAddressMainnet,
+	testnet: setIdbBtcAddressTestnet
+};
+
 const loadBtcAddress = async ({
 	tokenId,
 	network
 }: {
-	tokenId: typeof BTC_MAINNET_TOKEN_ID;
-	network: BitcoinNetwork;
+	tokenId: TokenIdBtcPublicNetwork;
+	network: BtcPublicNetwork;
 }): Promise<ResultSuccess> =>
 	loadTokenAddress<BtcAddress>({
 		tokenId,
 		getAddress: (identity: OptionIdentity) =>
 			getBtcAddress({
 				identity,
-				network: network === 'testnet' ? { testnet: null } : { mainnet: null }
+				network: bitcoinNetworkMapper[network]
 			}),
-		setIdbAddress: setIdbBtcAddressMainnet,
-		addressStore: btcAddressMainnetStore
+		setIdbAddress: idbBtcAddressMapper[network],
```

Comment:
Would it be interesting to merge `idbBtcAddressMapper` and `bitcoinStoreMapper` together?

Code wise we are fine with current implementation as we are using `network` to index the two but, technically there is no restriction so someday it might happens that one use `idbBtcAddressMapper` with `mainnet` and `bitcoinStoreMapper` with `testnet` or the contrary. You know what I mean?

```
...something[network]

something = {[network]: {addressMapper, storeMapper}}
```

Reply:
I think that if it starts to differ so much, maybe it's worth not using the same `loadBtcToken` function and create different ones.

Reply:
I'm no sure I conveyed my idea clearly and therefore not sure you understood my feedback so here's a PR: https://github.com/dfinity/oisy-wallet/pull/2302